### PR TITLE
refactor(core): Extract webhook context out of NodeExecutionFunctions (no-changelog)

### PR DESCRIPTION
**Pull request by netroy**
## Summary

This PR extracts Webhook context object out of NodeExecutionFunctions, and adds unit tests for most of this code.

## Related Linear tickets, Github issues, and Community forum posts

CAT-279

## Review / Merge checklist

- [x] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) 
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [x] Tests included. 
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

### CODE REVIEW

File: packages/core/src/node-execution-context/webhook-context.ts [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+	async getCredentials<T extends object = ICredentialDataDecryptedObject>(type: string) {
+		return await getCredentials<T>(this.workflow, this.node, type, this.additionalData, this.mode);
+	}
+
+	getBodyData() {
+		if (this.additionalData.httpRequest === undefined) {
+			throw new ApplicationError('Request is missing');
+		}
+		return this.additionalData.httpRequest.body as IDataObject;
+	}
```
Comment (tomi) [inline packages/core/src/node-execution-context/webhook-context.ts:81]: This keeps repeating. We could move it to a separate function:

```suggestion
    const httpRequest = this.assertHttpRequest()
		return httpRequest.body as IDataObject;
	}
	
	assertHttpRequest() {
		if (this.additionalData.httpRequest === undefined) {
			throw new ApplicationError('Request is missing');
		}
		
		return this.additionalData.httpRequest
	}
```

**Review by tomi — APPROVED** [review]
LGTM, one stylistic suggestion

File: packages/core/src/node-execution-context/webhook-context.ts (hunk 1 above)
Comment (netroy) [reply packages/core/src/node-execution-context/webhook-context.ts:81]: I'll make this change in the next PR 🙏🏽

Comment (netroy) [reply packages/core/src/node-execution-context/webhook-context.ts:81]: addressed here https://github.com/n8n-io/n8n/pull/11461/commits/ee1144c76b640680c5edc3f4448d7b5d878b7f3b

**Comment by janober** [comment]
Got released with [`n8n@1.67.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.67.0)

**Pull request merged**